### PR TITLE
Create Cloud SQL db programmatically if it does not exist.

### PR DESCRIPTION
**Pull request by keyurva**
* To support versioning, we'll be creating databases on cloud sql instances programmatically.
* This PR adds support for this in preparation of the larger versioning effort.
* The implementation is backward compatible since it creates a db only if one does not already exist.
* Tested locally by running the db script which created [this db](https://screenshot.googleplex.com/AKLsZXqUXVRBob3) programmatically.

### CODE REVIEW

File: simple/stats/db.py [hunk 1]
```diff
@@ -373,10 +373,16 @@ def commit_and_close(self):
       create_file_handler(self.db_file_path, is_dir=False).write_bytes(local_db)
 
 
-_CLOUD_MY_SQL_CONNECT_PARAMS = [
-    CLOUD_MY_SQL_USER, CLOUD_MY_SQL_PASSWORD, CLOUD_MY_SQL_DB
+# Parameters needed to connect to a Cloud SQL instance.
+_CLOUD_MY_SQL_INSTANCE_CONNECT_PARAMS = [
+    CLOUD_MY_SQL_USER, CLOUD_MY_SQL_PASSWORD
 ]
```
Comment (pradh) [inline simple/stats/db.py:379]: super nit: vertical spaces between (comment, variable) pairs?

File: simple/stats/db.py [hunk 2]
```diff
@@ -397,6 +406,34 @@ def __init__(self, db_params: dict[str, str]) -> None:
...
+                             db_params: dict[str, str]) -> None:
+    kwargs = {
+        param: db_params[param]
+        for param in _CLOUD_MY_SQL_INSTANCE_CONNECT_PARAMS
+    }
+    db_name = db_params[CLOUD_MY_SQL_DB]
+    logging.info(
+        "Connecting to Cloud MySQL instance '%s' to check existence of DB '%s'.",
+        db_params[CLOUD_MY_SQL_INSTANCE], db_name)
+    with connector.connect(db_params[CLOUD_MY_SQL_INSTANCE], "pymysql",
```
Comment (pradh) [inline simple/stats/db.py:419]: what is "pymysql", comment?

File: simple/stats/db.py [hunk 3]
```diff
@@ -397,6 +406,34 @@ def __init__(self, db_params: dict[str, str]) -> None:
...
+                           **kwargs) as conn:
+      with conn.cursor() as cursor:
+        # Check if the database already exists.
+        cursor.execute(
+            f"SELECT COUNT(*) FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = '{db_name}'"
+        )
+        exists = cursor.fetchone()[0] > 0
+
+        if not exists:
```
Comment (pradh) [inline simple/stats/db.py:428]: nit: for readability, maybe a helper `_db_exists(cursor)`?

File: simple/stats/db.py [hunk 4]
```diff
@@ -397,6 +406,34 @@ def __init__(self, db_params: dict[str, str]) -> None:
...
+  def _maybe_create_database(connector: Connector,
+                             db_params: dict[str, str]) -> None:
+    kwargs = {
+        param: db_params[param]
+        for param in _CLOUD_MY_SQL_INSTANCE_CONNECT_PARAMS
+    }
+    db_name = db_params[CLOUD_MY_SQL_DB]
+    logging.info(
+        "Connecting to Cloud MySQL instance '%s' to check existence of DB '%s'.",
+        db_params[CLOUD_MY_SQL_INSTANCE], db_name)
```
Comment (pradh) [inline simple/stats/db.py:418]: nit: maybe use a tmp var in place of two map refs

**Review by pradh — APPROVED** [review]
A few nits, that you should feel free to ignore :)

File: simple/stats/db.py (hunk 1 above)
Comment (keyurva) [reply simple/stats/db.py:379]: Done.

File: simple/stats/db.py (hunk 4 above)
Comment (keyurva) [reply simple/stats/db.py]: Done.

File: simple/stats/db.py (hunk 2 above)
Comment (keyurva) [reply simple/stats/db.py]: It's the db driver. Added comment.

File: simple/stats/db.py (hunk 3 above)
Comment (keyurva) [reply simple/stats/db.py]: Done.

**Review by keyurva — COMMENTED** [review]
Thanks for the review.

**Pull request merged**